card/mod/05_standard/set/all/links.rb

```diff
@@ -31,7 +31,11 @@ def web_link href, opts={}
   # link to a specific card
   def card_link name, opts={}
     text = (opts.delete(:text) || name).to_name.to_show @context_names
-    name = Card[name].cardname if Symbol===name
+    name = case name
+           when Symbol then Card[name].cardname
+           when Card   then name.cardname
+           else             name
+           end
```

Comment:
The codename cache is just a mapping between codenames and ids.  If you want the name you have to use fetch.  Otherwise the codename cache would have to get involved in cache clearing for name changes, which would suck.